src/RFC-0306_DANTemplateRegistration.md

```diff
@@ -0,0 +1,173 @@
+| build_image | Url       | A docker build environment used to build the binary  |
+| repo_link   | Url       | A public link to the source code repository          |
+| commit_hash | SHA2 hash | The commit hash of the code used to build the binary |
+
+The `build_image` field SHOULD contain a link to a publicly-accessible docker image that contains the exact build
+environment used to build the binary. The build environment refers to the specific compiler and OS used. Typically,
+this will be a docker image with a specific version of the rust compiler, LLVM and the `wasm-unknown-unknown` target
+provided by the Tari community.
+
+Anyone wishing to execute the template MAY build the binary themselves and compare checksums.
+It's worth noting that identical build environments do not guarantee deterministic builds.
+If you're curious about the kinds of issues encountered with deterministic builds using the rust language,
+[read this post](https://dev.to/gnunicorn/hunting-down-a-non-determinism-bug-in-our-rust-wasm-build-4fk1).
+
+## Obtaining the Template Binary
+
+The [contract definition] specifies the [binary_checksum] for each template. Once the [validator node] has been
+assigned a contract via the [contract definition], the [validator node] performs the following actions to obtain a template:
+
+1. It scans the blockchain for a template matching the `binary_checksum`;
+2. it downloads the WASM binary and verifies the checksum and stores the binary and associated registration [UTXO] metadata;
+3. a validator node operator may choose to build the binary from source as per `BuildInfo` and use the resulting binary;
+
+In the event that a new validator is added to [contract constitution], but the original template registration is unavailable,
+the [validator node] SHOULD make the template available to new committee members. New committee members SHOULD confirm the checksum
+with a 2/3 majority of the validator committee to ensure that the correct copy is received.
+
+## Spending the Template Registration UTXO
+
+These cases apply to spending the template registration [UTXO]:
+1. The author MAY spend into another template registration [UTXO]
+* This effectively withdraws (yanks) the previous version of the template.
+* The template name and author SHOULD be identical. This MAY be enforced by covenant.
+* The template version SHOULD be incremented.
+* Any live contracts SHOULD upgrade their definitions to run the new template.
+2. The author MAY spend to a "vanilla" [UTXO] to reclaim their deposit.
+* This effectively withdraws (yanks) the template.
+* The template MAY remain in use on existing contracts. In fact, anyone may now re-register the template.
+
+Alternatives:
+* Consensus rule that prevents spending of the template while used by other contracts
+* The contract definition may have copy the template binary_checksum etc. Validator nodes may mirror the code for new VNs
+
+## Upgrading a Template
```

Comment:
We should spec this out more as this could be troublesome if they upgrade out of sync